### DISCUSSION TITLE
Please help me converting for my LED-Strip

### DISCUSSION THREAD

**Discussion by consoleman** [RavenSystem/esp-homekit-devices#2023] [MEPLHAA Scripts]
I have a BlitzWolf LED Strip and I didn't manage to convert the script, please help.

{"c":{"b":[{"g":0,"t":5}]},"a":[{"t":30,"g":[4,12,14,5],"ld":[0,1,0,1],"b":[{"g":0}]}]}

And also for my Sonoff 2 Gang touch switch:

{"c":{"l":13,"b":[{"g":10,"t":5}]},"a":[{"0":{"r":[{"g":12}]},"1":{"r":[{"g":12,"v":1}]},"b":[{"g":0,"t":0}],"es":[{"0":{"r":[{"g":5}]},"1":{"r":[{"g":5,"v":1}]},"b":[{"g":9,"t":0}]}]}]}

**Comment by consoleman** [comment]
Nobody?

**Reply by xrust83** [reply]
https://github.com/RavenSystem/esp-homekit-devices/wiki/Migration-Guide-for-HAA-V12-Merlin#pwm

**Reply by xrust83** [reply]
https://github.com/RavenSystem/esp-homekit-devices/wiki/Devices-Database

Itead 2CH

**Reply by consoleman** [reply]
Thanks for the links, but I cannot manage to get it to work.
Simply I do not understand the way to generate the script

**Reply by xrust83** [reply]
There is a ready-made butt, it is necessary to rewrite your gpio in the same sequence

**Reply by xrust83** [reply]
{"c":{"io":[[[4,14],7],[[12,5],7,0,0,2],[[0],6,1]],"b":[{"g":0,"t":5}]},"a":[{"t":30,"g":[4,12,14,5],"b":[{"g":0}]}]}

**Reply by xrust83** [reply]
{"c":{"io":[[[5,12,13],2],[[0,9,10],6]],"l":13,"b":[{"g":10,"t":5}]},"a":[{"0":{"r":[{"g":12}]},"1":{"r":[{"g":12,"v":1}]},"b":[{"g":0,"t":0}],"es":[{"0":{"r":[{"g":5}]},"1":{"r":[{"g":5,"v":1}]},"b":[{"g":9,"t":0}]}]}]}
